packages/create-react-app-demo/src/recipe/page.tsx

```diff
@@ -102,9 +117,18 @@ export default function RecipeWrapper() {
               </Recipe>
             }
           >
-            <ChatCompletion>
-              <UserMessage>Give me a recipe for {query}.</UserMessage>
-            </ChatCompletion>
+            {recipe}
+            {'\n'}
+            Now here's a link to the banner URL for the recipe:{' '}
+            <ImageGen>
+              Generate an image for a recipe name
+              <ChatCompletion>
+                <UserMessage>
+                  In two to three sentences, describe how the following recipe would look like when prepared by a chef:
+                  {recipe}
```

Comment:
maybe want to try doing this based on {query}, to avoid introducing a 3rd pass?

Reply:
Yeah I did that originally and it works easily.
The main issue is image not matching the recipe. The query along is usually not descriptive enough.

At the very least, I should change the default from "beans" to an actual dish name 😅